Create p1 Algolia site search package

B77Mills opened the pull request:
This is a port from the following [PR](https://github.com/parameter1/ac-business-media-websites/pull/277) to move the package to base-cms.  

This package will require env variable for the following 
 - ALGOLIA_API_KEY
 - ALGOLIA_APP_ID
 - ALGOLIA_DEFAULT_INDEX
 
 onStart the following middleware will also have to be accounted for 
``` 
const algoliaSiteSearch = require('@parameter1/marko-web-algolia-site-search/middleware/algolia');
const algoliaSiteSearchPaginated = require('@parameter1/marko-web-algolia-site-search/middleware/paginated');
 
 onStart: (app) => {
    app.use(algoliaSiteSearch());
    app.use(algoliaSiteSearchPaginated());
  },
  ```

B77Mills committed:
Move Algoia site search to P1 package

Move from the ACBM org to the P1 dep packages.

https://github.com/parameter1/ac-business-media-websites/pull/277

Once merged I will have to update this PR to pull in from P1 packages.

B77Mills committed:
update repository path and add description

zarathustra323 commented on packages/marko-web-algolia-site-search/components/index.marko:
```diff
@@ -0,0 +1,153 @@
+import { getAsObject, getAsArray, get } from "@parameter1/base-cms-object-path";
```
This is probably too site specific to be included in a generic package... really this package should only provide the middleware and the query wrapper components (for doing the algolia query + the base cms query)

zarathustra323 commented on packages/marko-web-algolia-site-search/components/nodes/content-list.marko:
```diff
@@ -0,0 +1,141 @@
+import { getAsObject, get } from "@parameter1/base-cms-object-path";
```
Same here... this should be handled by the site

zarathustra323 commented on packages/marko-web-algolia-site-search/components/pagination-controls.marko:
```diff
@@ -0,0 +1,53 @@
+import { cleanPath } from "@parameter1/base-cms-utils";
```
This could be included, but it should be a "functional" component only -- meaning, the site should determine how the controls are rendered (via a `input.renderBody` call) and the package component merely provides the wrapping logic

zarathustra323 commented on packages/marko-web-algolia-site-search/graphql/fragments/content-list.js:
```diff
@@ -0,0 +1,67 @@
+const gql = require('graphql-tag');
```
Same for fragments -- these should be provided by the websites, as each site will likely have different field requirements depending on their layouts

zarathustra323 commented on packages/marko-web-algolia-site-search/package.json:
```diff
@@ -0,0 +1,29 @@
+{
+  "name": "@ac-business-media/marko-web-algolia-site-search",
+  "description": "Algoli site search components for BaseCMS websites.",
```
Missing the `a` in Alogila :)

zarathustra323 commented on packages/marko-web-algolia-site-search/package.json:
```diff
@@ -0,0 +1,29 @@
+{
+  "name": "@ac-business-media/marko-web-algolia-site-search",
+  "description": "Algoli site search components for BaseCMS websites.",
+  "version": "0.0.0",
```
This version number should be set to the current value found in the root `lerna.json` file

B77Mills committed:
Make pagination controls use renderBody logic

B77Mills committed:
update query call to use render body

Add ability to pass it the list of filters to apply instead of doing that in core and utilize renderBody to return values to template that is calling it.

B77Mills committed:
Remove none core components/fragments/utils

Remove all site specific things from core package.  Should leave that in site specific spots to figure out what filters to pass and how to display facet layouts.  Can look at adding this to the default theme if need be for global use, but this should only worry about getting info on how to query algolia and what values to pass back to the site.

B77Mills committed:
fix spelling error and version #
